pkg/minishift/provisioner/utils.go

```diff
@@ -265,3 +266,19 @@ func configureAuth(p *BuildrootProvisioner) error {
 
 	return nil
 }
+
+func doFeatureDetection(p provision.Provisioner) error {
+	if out, err := p.SSHCommand(fmt.Sprintf("test -f %s && echo '1' || echo '0' ", "/usr/local/bin/minishift-set-ipaddress")); err != nil {
+		return err
+	} else {
+		if strings.Trim(out, "\n") == "1" {
+			minishiftConfig.InstanceConfig.SupportsNetworkAssignment = true
```

Comment:
IMO it is very unfortunate to make calls to static _minishiftConfig.InstanceConfig_. This is something we want to get away from if anything. This way one creates dependencies which are hard to maintain and test.

Reply:
This is something that needs to be addressed in a separate issue:

https://github.com/minishift/minishift/blob/77422c5cea568cfc4912942ba2ff5df37c40e9ae/pkg/minishift/provisioner/minishift_provisioner.go#L78-L79

Modifications to config are not part of this change, see #1796